include/openenclave/internal/thread.h

```diff
@@ -54,17 +54,34 @@ bool oe_thread_equal(oe_thread_t thread1, oe_thread_t thread2);
 
 typedef uint32_t oe_once_t;
 
+/* Match the definitions of corelibc/pthread.h */
+#define OE_MUTEX_NORMAL 0
+#define OE_MUTEX_DEFAULT 0
+#define OE_MUTEX_RECURSIVE 1
```

Comment:
Do we ever create a oe_mutex without an initializer? If so, then previously it will be initialized to zero (if a global variable). Since OE_MUTEX_RECURISVE is 1 and not 0, with this PR such mutexes will no longer be recursive.

Reply:
I've updated the pthread implementation to differentiate it from oe_mutex implementation